client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/internal/dft/DefaultClientTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client.internal.dft;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DefaultClientTest {
+
+    @Test
+    void testAcceptsURL() {
+        var client = new DefaultClient();
+        assertThat(client.acceptsURL("pal://localhost")).isTrue();
+        assertThat(client.acceptsURL("jdbc://localhost")).isFalse();
```

Comment:
This seems to just add a needless check that the URL starts with some scheme.
What about when palisade or frs might be https and wss?
What about when the data-service is hdfs or something weird? (s3 is a likely case too)
If you point it to the wrong cluster, that's a user error, but if you forget the `pal` at the start...? what are we actually gaining from this?

Reply:
This is along the line of JDBC (unofficial scheme but widely supported). The actual protocols used to talk to each of the underlying services is not part of the URL. They can be configured via properties. HDFS would be represented by a URI (e.g. hdfs://{NameNodeHost}:8020/). But this support would have to be added via a plugin. These could then be looked up by the scheme (https://en.wikipedia.org/wiki/List_of_URI_schemes).